doc/source/WorkingWithThePythonScriptingAPI/index.md

```diff
@@ -0,0 +1,30 @@
+# Working with the Python Scripting API #
+
+With its core architecture built from a combination of C++ and Python, Gaffer ships with a convenient Python scripting API. A significant portion of the project has hooks in Python: nodes are exposed to Python in the main application, the node graph file formats are Python scripts, much of the GUI is written in Python, and the various Gaffer applications are configured by Python scripts. Through the main application's built-in _Python Editor_, you can interactively build and modify the node graph, test expressions, and inspect data. What's more, with a modicum of knowledge in Python and the Gaffer architecture, you can configure and customize each Gaffer application to suit your studio or pipeline needs.
+
+> Important :
+> With regards to node graphs, do not confuse the use of the scripting API with expressions inside Expression nodes. The scripting API can be applied in the main application to interactively modify, inspect, test, and debug a node graph. Expressions, on the other hand, are bits of code stored on Expression nodes **inside** a graph, that manipulate data as part of the graph's function.
```

Comment:
> Expression nodes **inside** a graph, that manipulate data as part of the graph's function

Expressions don't directly manipulate data, they're connected to Plugs to determine their value during compute. How about: 

"... that allow you to dynamically change a plugs value."

... or something to that effect?

Reply:
Sure, let's just make it a little clearer:

> ... that allow you to dynamically change plug values during the graph's execution.

On this topic: When using the term "data" to relate "information that flows through the graph," do we want a hard separation between data that results from querying `out` plugs with context, and "actual" 2D/3D data, as it were? Do we want to separate them for developers?

Reply:
There is no difference between the two! The 'actual' data _is_ just data from plugs queried with a context (hence some of my previous comments about not talking about 'data' in certain places).

Reply:
The re-world looks good though, thanks!

Reply:
Just thinking about the above, maybe the confusion is that internally, Expressions use python/OSL to derive the value of _their_ out plugs, these plugs are then connected to the 'target' plugs on some other Node. Do during compute when an affected node queries the value of its plug to determine what it should do, it will result in the value being pulled from the connected expression output - just as any other plug-plug connection. If that makes sense?